### PR TITLE
add hw3

### PR DESCRIPTION
**Pull request by RiinaVi**

### CODE REVIEW

**Comment by AndriiPolishko** [comment]
Do you merge not into main in purpose?🙂

File: src/app.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import express, { Request, Response } from 'express';
+import dotenv from 'dotenv';
+import cors from 'cors';
+import bodyParser from 'body-parser';
+import { ormconfig as dataSource } from './ormconfig';
+
```
Comment (AndriiPolishko) [inline src/app.ts]: obsolete empty line

File: src/app.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import schedule from 'node-schedule';
+
+import router from './routes';
+import sendRateToAllEmails from './jobs/sendRateToAllEmails';
+import responseMessages from './constants/responseMessages';
+
+const { SOMETHING_WENT_WRONG } = responseMessages;
+// every day at 10:00
+const SCHEDULING_TIME = '00 10 * * *';
```
Comment (AndriiPolishko) [inline src/app.ts]: could you please give a more self explanatory name

File: src/tests/getUnsubscribedURL.test.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+  test('should return URL with the passed host and port', () => {
+    const email = 'test@example.com';
+    const host = 'test.com';
+    const port = 5000;
+    const unsubscribeURL = getUnsubscribeURL(email, host, port);
+
+    expect(unsubscribeURL).toEqual(
+      `http://${host}:${port}/unsubscribe?email=${email}`,
+    );
```
Comment (AndriiPolishko) [inline src/tests/getUnsubscribedURL.test.ts:21]: since you test same function woudn't it be a good idea to write one test that result equals to `http://${host}:${port}/unsubscribe?email=${email}`, ?

File: src/tests/integration/EmailSendingService.test.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
+/* eslint-disable */
```
Comment (AndriiPolishko) [inline src/tests/integration/EmailSendingService.test.ts]: Please remove :)

File: src/tests/integration/RateSourceService.test.ts [hunk 5]
```diff
@@ -0,0 +1,80 @@
+/* eslint-disable */
```
Comment (AndriiPolishko) [inline src/tests/integration/RateSourceService.test.ts]: please remove :)

File: src/tests/integration/RateSourceService.test.ts [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+      },
+    };
+    fetchStub.resolves(jsonErrorResponseFromThirdPartyService);
+
+    const response = await request(app).get('/rate');
+    expect(response.status).toBe(429);
+    expect(response.body).toStrictEqual({
+      error: `${errorResponseFromThirdPartyService.meta.error_type}: ${errorResponseFromThirdPartyService.meta.error_detail}`,
+    });
+  });
```
Comment (AndriiPolishko) [inline src/tests/integration/RateSourceService.test.ts:55]: not sure if this tests do tests anything since the response is mocked

**Review by AndriiPolishko — CHANGES_REQUESTED** [review]
Cool!

Please remove /* eslint-disable */ from all files

File: src/tests/integration/RateSourceService.test.ts (hunk 6 above)
Comment (RiinaVi) [reply src/tests/integration/RateSourceService.test.ts:55]: Actually this is the test that found a huge problem within my app, because I previously didn't handle errors form third-party service properly and it crashed my app

So now I'm passing status codes and errors to my endpoints

And when I add another third party services - I need these tests to work, I hope that makes sense :)

File: src/tests/getUnsubscribedURL.test.ts (hunk 3 above)
Comment (RiinaVi) [reply src/tests/getUnsubscribedURL.test.ts:21]: Well, we were supposed to write tests with edge cases ¯\_(⊙︿⊙)_/¯
And this function has some non-required arguments

File: src/tests/load/unsubscribe.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+  stages: [
+    { duration: '30s', target: 20 },
+    { duration: '1m', target: 50 },
+    { duration: '30s', target: 0 },
+  ],
+};
+
+export default function () {
+  const email = 'riinavi86@gmail.com';
+  const res = http.post(`http://localhost:8000/unsubscribe?email=${email}`);
```
Comment (mykola-koval-liven) [inline src/tests/load/unsubscribe.js]: Всі тести дивляться на 8000-й порт, якщо він буде зайнятий як бути?

**Review by mykola-koval-liven — APPROVED** [review]
Чудово що відбранчовано від попереднього PR. Часто в студентів з цим виникає проблема
